src/cgame/default/viewcontrollers/CreateServerViewController.c

```diff
@@ -30,24 +30,73 @@
 #pragma mark - Actions and delegate callbacks
 
 /**
+ * @brief Select gameplay mode
+ */
+static void selectGameplay(Select *select, Option *option) {
+	if (option->value == (ident) 0) {
+		cgi.CvarSet(g_gameplay->name, "default");
+	} else if (option->value == (ident) 1) {
+		cgi.CvarSet(g_gameplay->name, "deathmatch");
+	} else if (option->value == (ident) 2) {
+		cgi.CvarSet(g_gameplay->name, "instagib");
+	} else if (option->value == (ident) 3) {
+		cgi.CvarSet(g_gameplay->name, "arena");
+	} else if (option->value == (ident) 4) {
+		cgi.CvarSet(g_gameplay->name, "duel");
+	}
+}
+
+/**
+ * @brief Select teamplay mode
+ */
+static void selectTeamsplay(Select *select, Option *option) {
+	if (option->value == (ident) 0) {
+		cgi.CvarSet(g_teams->name, "0");
+		cgi.CvarSet(g_ctf->name, "0");
+	} else if (option->value == (ident) 1) {
+		cgi.CvarSet(g_teams->name, "1");
+		cgi.CvarSet(g_ctf->name, "0");
+	} else if (option->value == (ident) 2) {
+		cgi.CvarSet(g_teams->name, "0");
+		cgi.CvarSet(g_ctf->name, "1");
+	}
+}
+
+/**
  * @brief ActionFunction for the Create button.
  */
 static void createAction(Control *control, const SDL_Event *event, ident sender, ident data) {
 
 	CreateServerViewController *this = (CreateServerViewController *) sender;
 
+	char maplist[MAX_STRING_CHARS] = "";
+	char firstmap[MAX_QPATH] = "";
+
+
 	GList *selectedMaps = $(this->mapList, selectedMaps);
 	for (const GList *list = selectedMaps; list; list = list->next) {
 
-		g_map_list_map_t *map = cgi.Malloc(sizeof(g_map_list_map_t), MEM_TAG_CGAME);
+		char name[MAX_QPATH];
+		g_strlcpy(name, (const char *) Basename(list->data), sizeof(name));
+		StripExtension(name, name);
 
-		g_strlcpy(map->name, (const char *) list->data, sizeof(map->name));
-		printf("selected %s\n", map->name);
+		if (maplist[0] == '\0') {
+			g_strlcpy(firstmap, name, sizeof(firstmap));
+		}
 
-		//mapList = g_list_append(mapList, map);
+		g_strlcpy(maplist, va("%s%s ", maplist, name), sizeof(maplist));
```

Comment:
Since we are in Objectively, a MutableString would make this much easier -- but we can clean that up later. No need to hold up the PR.